### DISCUSSION TITLE
The Operation in FRD systems

### DISCUSSION THREAD

**Discussion by JinmingChe** [python-control/python-control#611] [Q&A]
Hi, when I use python to implement a function in matlab, how do I calculate the inverse of the FRD system and the calculation between the FRD system and the constant? In matlab, it can be implemented directly through the ‘inv’ and ‘+,/, -’ operators.

**Comment by murrayrm** [comment]
What do you mean by "inverse"?  If you just want to compute something like `S = 1 / (1 + P * C)` then you just use that syntax.  All of the usual operators work for FRD systems.  (You can also use `series`, `parallel`, and `feedback`.)

**Reply by JinmingChe** [reply]
Something  like L = inv(S)-1 in matlab, which S is the FRD system.  I try to use the inv function in np.linalg (np.linalg.inv(S) in python), but it doesn't work.

**Reply by murrayrm** [reply]
The `inv` function is not implemented.  If you are trying to doing block diagram algebra, it is better to just use `feedback`.  For example: `S = ct.feedback(Id, P * C)` where `Id` is a system implementing the identity I/O system in the appropriate dimensions.  (You have to create the `Id` function manually since nobody has implemented broadcasting of constants yet.)

**Reply by JinmingChe** [reply]
May I ask what p and c （in S = ct.feedback(Id, P * C)） stand for?

**Reply by murrayrm** [reply]
Process and controller transfer function.  Here's is a working example using FRDs:
```
import control as ct
import numpy as np

omega = np.logspace(-1, 1, 10)
P = ct.frd(ct.rss(3, 2, 2), omega)
C = ct.frd(ct.rss(2, 2, 2), omega)
Id = ct.frd(ct.ss([], [], [], np.eye(2)), omega)
S = ct.feedback(Id, P * C)
```
